### PR TITLE
Rename any reference of cadence to temporal

### PR DESCRIPTION
**Pull request by samarabbas**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
-# Go framework for Cadence [![Build Status](https://badge.buildkite.com/e7241785444519bdfd1defc68839fd19a89c15adb3477c73f7.svg?theme=github&branch=master)](https://buildkite.com/uberopensource/cadence-go-client) [![Coverage Status](https://coveralls.io/repos/uber-go/cadence-client/badge.svg?branch=master&service=github)](https://coveralls.io/github/uber-go/cadence-client?branch=master) [![GoDoc](https://godoc.org/go.temporal.io/temporal?status.svg)](https://godoc.org/go.temporal.io/temporal)
+# Go framework for Temporal [![Build Status](https://badge.buildkite.com/ce6df3b1a8b375270261ae70fb2d2756af298fef3a0dac4d20.svg?theme=github&branch=master)](https://buildkite.com/temporal/temporal-go-client) [![GoDoc](https://godoc.org/go.temporal.io/temporal?status.svg)](https://godoc.org/go.temporal.io/temporal)
```
Comment (mfateev) [inline README.md]: May be change to:  "Temporal Go SDK"

File: activity/doc.go [hunk 2]
```diff
@@ -19,15 +19,15 @@
...
+Package activity contains functions and types used to implement Temporal activities.
 
 The activity is an implementation of a task to be performed as part of a larger workflow. There is no limitation of
 what an activity can do. In the context of a workflow, it is in the activities where all operations that affect the
 desired results must be implemented.
 
 Overview
 
-The client library for Cadence does all the heavy lifting of handling the async communication between the Cadence
+The client library for Temporal does all the heavy lifting of handling the async communication between the Temporal
```
Comment (mfateev) [inline activity/doc.go]: Temporal Go SDK

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,14 +1,12 @@
-# Developing cadence-client
+# Developing temporal-client
 
-This doc is intended for contributors to `cadence-client` (hopefully that's you!)
+This doc is intended for contributors to `temporal-client` (hopefully that's you!)
 
 **Note:** All contributors also need to fill out the [Uber Contributor License Agreement](http://t.uber.com/cla) before we can merge in any of your changes
```
Comment (mfateev) [inline CONTRIBUTING.md]: Remove or add reference to Temporal one.

File: README.md [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 
-[Cadence](https://github.com/uber/cadence) is a distributed, scalable, durable, and highly available orchestration engine we developed at Uber Engineering to execute asynchronous long-running business logic in a scalable and resilient way.
+[Temporal](https://github.com/temporalio/temporal) is a distributed, scalable, durable, and highly available orchestration engine we developed at Uber Engineering to execute asynchronous long-running business logic in a scalable and resilient way.
```
Comment (mfateev) [inline README.md]: Rephrase the "we developed at Uber Engineering"

File: CONTRIBUTING.md (hunk 3 above)
Comment (samarabbas) [reply CONTRIBUTING.md]: Will do this in next change

File: README.md [hunk 5]
```diff
@@ -1,15 +1,15 @@
-# Go framework for Cadence [![Build Status](https://badge.buildkite.com/e7241785444519bdfd1defc68839fd19a89c15adb3477c73f7.svg?theme=github&branch=master)](https://buildkite.com/uberopensource/cadence-go-client) [![Coverage Status](https://coveralls.io/repos/uber-go/cadence-client/badge.svg?branch=master&service=github)](https://coveralls.io/github/uber-go/cadence-client?branch=master) [![GoDoc](https://godoc.org/go.temporal.io/temporal?status.svg)](https://godoc.org/go.temporal.io/temporal)
+# Temporal GO SDK [![Build Status](https://badge.buildkite.com/ce6df3b1a8b375270261ae70fb2d2756af298fef3a0dac4d20.svg?theme=github&branch=master)](https://buildkite.com/temporal/temporal-go-client) [![GoDoc](https://godoc.org/go.temporal.io/temporal?status.svg)](https://godoc.org/go.temporal.io/temporal)
```
Comment (alexshtin) [inline README.md]: I think we agreed to rename "go client" to "Go SDK" everywhere including repo name (should be `temporalio/temporal-go-sdk`). Let's do it on separate PR though.
nit: Should be "Go SDK", not "GO SDK".

File: internal/common/metrics/constants.go [hunk 6]
```diff
@@ -22,78 +22,78 @@ package metrics
...
-	WorkerStartCounter = CadenceMetricsPrefix + "worker-start"
-	PollerStartCounter = CadenceMetricsPrefix + "poller-start"
+	WorkerStartCounter = TemporalMetricsPrefix + "worker-start"
+	PollerStartCounter = TemporalMetricsPrefix + "poller-start"
 
-	CadenceRequest        = CadenceMetricsPrefix + "request"
-	CadenceError          = CadenceMetricsPrefix + "error"
-	CadenceLatency        = CadenceMetricsPrefix + "latency"
-	CadenceInvalidRequest = CadenceMetricsPrefix + "invalid-request"
+	TemporalRequest        = TemporalMetricsPrefix + "request"
```
Comment (alexshtin) [inline internal/common/metrics/constants.go]: There are few places like this where we don't actually need `Temporal` or `Cadence` to be the part of the var/func/object name. Also there are a lot of "temporal" in comments where it can be safely removed (like "temporal server" => "server")

File: workflow/doc.go [hunk 7]
```diff
@@ -393,13 +393,13 @@ SideEffect function any other way than through its recorded return value.
 
 Query API
 
-A workflow execution could be stuck at some state for longer than expected period. Cadence provide facilities to query
-the current call stack of a workflow execution. You can use cadence-cli to do the query, for example:
+A workflow execution could be stuck at some state for longer than expected period. Temporal provide facilities to query
+the current call stack of a workflow execution. You can use temporal-cli to do the query, for example:
 
-	cadence-cli --domain samples-domain workflow query -w my_workflow_id -r my_run_id -qt __stack_trace
+	temporal-cli --domain samples-domain workflow query -w my_workflow_id -r my_run_id -qt __stack_trace
```
Comment (alexshtin) [inline workflow/doc.go]: image

Comment (alexshtin) [reply workflow/doc.go]: I mean CLI will be `temporal` not `temporal-cli`. Actually I believe it needs to be renamed to `tctl`.

Comment (samarabbas) [reply workflow/doc.go]: Let's iterate over this as separate PR.

File: internal/common/metrics/constants.go (hunk 6 above)
Comment (samarabbas) [reply internal/common/metrics/constants.go]: Agreed.  I'll do that as a separate PR.

File: README.md (hunk 5 above)
Comment (samarabbas) [reply README.md]: Agreed.
